src/NUnitFramework/framework/Internal/TestNameGenerator.cs

```diff
@@ -450,65 +473,71 @@ public FixedTextFragment(string text)
                 _text = text;
             }
 
-            public override string GetText(MethodInfo method, object[] args, bool argsAreDisplayNames)
+            public override string GetText(Test test, object[] args, bool argsAreDisplayNames)
             {
                 return _text;
             }
         }
 
         private class MethodNameFragment : NameFragment
         {
-            public override string GetText(MethodInfo method, object[] args, bool argsAreDisplayNames)
+            public override string GetText(Test test, object[] args, bool argsAreDisplayNames)
             {
+                var method = test.Method;
+                if (method == null) return null;
+
```

Comment:
Is there a case where test.Method should be null? Can we comment it, and if not, maybe `Guard.NotNull()` would be more explicit?

Reply:
Ah, presumably where Test is a TestFixture, so there's no method.

Reply:
Yes.

Reply:
Should we return an empty string instead of null? The only usage I see currently is a `StringBuilder.Append()` which is safe, but it might change in the future.